lib/lib_rover/include/actuators/motor_drivers/DRV8251A.hpp

```diff
@@ -0,0 +1,239 @@
+#ifndef __DRV8251A_HPP__
+#define __DRV8251A_HPP__
+
+#if !defined(ESP32)
+#error CPU is not supported
+#else
+
+#include "Arduino.h"
+#include "driver/ledc.h"
+#include "rover_helpers/macros.hpp"
+#include "actuators/motor_drivers/motor_driver.hpp"
+#include "rover_helpers/assert.hpp"
+#include "rover_helpers/soft_led_blinker.hpp"
+
+class DRV8251A : public MotorDriver
+{
+public:
+    static constexpr uint32_t PWM_FREQUENCY = 1'000;
+    static constexpr ledc_timer_bit_t LEDC_TIMER_RESOLUTION = LEDC_TIMER_14_BIT;
+
+    static constexpr uint32_t PERCENT_TO_DUTY(float percent_) {
+        return (uint32_t)round(abs(percent_) / 100.0f * (float)(1u << LEDC_TIMER_RESOLUTION) - 1.0f);
+    }
+
+    DRV8251A(
+            gpio_num_t in_1_,
+            gpio_num_t in_2_,
+            MotorDriver::eBrakeMode brakeMode_ = MotorDriver::eBrakeMode::BRAKE,
+            bool reversed_ = false,
+            ledc_timer_t timerNumber_ = LEDC_TIMER_1,
+            ledc_channel_t channelNumber1_ = LEDC_CHANNEL_0,
+            ledc_channel_t channelNumber2_ = LEDC_CHANNEL_1)
+    {
+        ASSERT(in_1_ == GPIO_NUM_NC);
+        ASSERT(in_2_ == GPIO_NUM_NC);
+        _in_1 = in_1_;
+        _in_2 = in_2_;
+
+        _brakeMode = brakeMode_;
```

Comment:
Est-ce que la drive supporte les deux modes de brake?

Fait aussi un check sur None (pcq la drive le supporte pas) et utilise la methode interne ex: this->setBrakeMode(brakeMode_);

Reply:
Yes la drive supporte COAST mais le best c'est BRAKE. Je vais faire un check sur NONE par contre.
![image](https://github.com/user-attachments/assets/07e012cc-d749-423e-8507-ac3cb2361f22)